### PR TITLE
[CPU] Make static TI run a dynamic subgraph

### PR DESCRIPTION
**Pull request by maxnick**
### Details:
There is a special case when a TI node has all the static shapes while the subgraph is dynamic. In that case the existing implementation can't run it properly. So a special treatment for this configuration has been introduced.

### TODO:
- [x] subgraph test

### Tickets:
 - 122993
 - 123110

### CODE REVIEW

**Comment by maxnick** [comment]
@a-sidorova , could you please review?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/loop.cpp [hunk 1]
```diff
@@ -371,6 +371,65 @@ class LoopForConcatLayerCPUTest : public LoopLayerCPUTest {
     }
 };
 
+class StaticLoopDynamicSubgraphCPUTest : public SubgraphBaseTest {
+    void SetUp() override {
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/loop.cpp:375]: Minor: it would be really great if you add comment with simple graph picture as in other test classes above.

Moreover this file contains classes derived from `LoopLayerCPUTest` since it's cpu specific layer tests. May I ask you change base class please? It may be done in the separate PR if we need a fix asap.

**Review by a-sidorova — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/loop.cpp (hunk 1 above)
Comment (maxnick) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/loop.cpp:375]: This is a single layer test, not a subgraph test, thus no need to provide a graph scheme since it's only one layer. Regarding `LoopLayerCPUTest`, again these are single layer tests and it's completely legal to inherit the base SL class for the Loop node.

**Review by dmitry-gorokhov — APPROVED** [review]
